src/analysis_app.py

```diff
@@ -221,45 +218,87 @@ def analyze(self, row_stream):
-            req['diacritization'] = self.params.diacritization
         if self.params.reference_date:
             req['referenceDate'] = self.params.reference_date
-        if self.params.full_analysis_output:
-            req['returnMentions'] = True
         return req
 
     def doc_batch_stream(self, row_stream):
         for rows in slice_stream(row_stream, self.params.doc_batch_size):
-            yield list(map(self.row_to_doc, rows))
+            yield [doc for row in rows for doc in self.row_to_docs(row)]
 
-    def row_to_doc(self, row):
+    def row_to_docs(self, row):
         def join_cols(columns):
             return '\n\n'.join(row[col] for col in columns if row[col])
 
-        doc = {
-            'id': json.dumps(list(row[id_col] for id_col in self.params.id_cols)),
-            'text': join_cols(self.params.text_cols)
+        ids = [row[id_col] for id_col in self.params.id_cols]
+        yield {
+            'id': json.dumps(['txt'] + ids),
+            self.sect_to_segm['text']: join_cols(self.params.txt_cols)
         }
-        if self.params.title_cols:
-            doc['title'] = join_cols(self.params.title_cols)
-        if self.params.lead_cols:
-            doc['lead'] = join_cols(self.params.lead_cols)
-        return doc
+        if self.params.pos_cols:
+            yield {
+                'id': json.dumps(['pos'] + ids),
+                self.sect_to_segm['positives']: join_cols(self.params.pos_cols)
+            }
+        if self.params.neg_cols:
+            yield {
+                'id': json.dumps(['neg'] + ids),
+                self.sect_to_segm['negatives']: join_cols(self.params.pos_cols)
+            }
+
+    def proc_batch_analysis(self, batch_analysis):
+        grouped = defaultdict(dict)
+        for doc_analysis in batch_analysis:
+            doc_type, *ids = json.loads(doc_analysis['id'])
+            grouped[tuple(ids)][doc_type] = doc_analysis
+
+        for ids, analysis_by_type in grouped.items():
+            doc_analysis = analysis_by_type['txt']
+            doc_analysis['id'] = json.dumps(list(ids))
+            self.proc_entities(doc_analysis['entities'], 'txt')
+            if 'pos' in analysis_by_type:
+                pos_analysis = analysis_by_type['pos']
+                self.proc_entities(pos_analysis['entities'], 'pos')
+                self.copy_analysis(pos_analysis, doc_analysis)
+            if 'neg' in analysis_by_type:
+                neg_analysis = analysis_by_type['neg']
+                self.proc_entities(neg_analysis['entities'], 'neg')
+                self.copy_analysis(neg_analysis, doc_analysis)
+            yield doc_analysis
+
+    def proc_entities(self, entities, doc_type):
+        feedback_entities = [e for e in entities if e['type'] in self.params.feedback_entities]
+        for ent in feedback_entities:
+            polarity = ent.get('sentiment', {}).get('polarity', 0)
+            if doc_type == 'pos' or (doc_type == 'txt' and polarity > 0):
```

Comment:
I think we should treat neutral polarity as positive. Hence, I'd use `(doc_type == 'txt' and polarity >= 0)`.

Reply:
2791c13